ypathgen/pathgen.go

```diff
@@ -761,15 +787,28 @@ func generateDirectorySnippet(directory *ygen.Directory, directories map[string]
 		}
 		goFieldName := goFieldNameMap[fieldName]
 
-		if es := generateChildConstructors(&methodBuf, directory, fieldName, goFieldName, directories, schemaStructPkgAccessor, pathStructSuffix, listBuilderKeyThreshold, generateWildcardPaths, simplifyWildcardPaths); es != nil {
+		fieldDirectory := directories[field.Path()]
+		childAccessor := ""
+
+		if fieldDirectory != nil {
```

Comment:
if fieldDirectory := directories[field.Path()]; fieldDirectory != nil {

Also I'm not so sure about having a general functionality to find out the `childPkgAccessor`. Might it be better to only do this handling at the root, because we know that's the only place we'll ever have this handling?

Reply:
fieldDirectory is used outside the scope of that if statement, can't make that change.

i could add a check for ygen.IsFakeRoot for this

Reply:
`fieldDirectory` is only used within the if statement right? where else is it being used?